perl/p5-net-apns-persistent/Portfile

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           perl5 1.0
+
+perl5.branches      26 28 30
+perl5.setup         Net-APNS-Persistent 0.02 ../by-authors/id/A/AU/AUFFLICK
+platforms           darwin
+maintainers         nomaintainer
+license             {Artistic-1 GPL}
+description         Send Apple APNS notifications over a persistent connection.
+
+long_description    Class to create a persistent connection to Apple's \
+    APNS servers.
+
+checksums           rmd160  0468bcf46c73739766332767ffc93a939c4cf2a2 \
+                    sha256  7984e4eaaa1edcddaa868762efeb36406f5ff5b9f0d2455f8c0b217e5429f397 \
+                    size    9755
+
+depends_run-append  \
```

Comment:
Dependencies should only be added if `{perl5.major} ne ""`. See existing p5-* ports for examples.

Reply:
I’m still confused about this. Why is lint throwing these errors with the current commit?

I thought that `${perl5.major}` was set?

If someone who knows how to write perl Portfiles wants to step in and just set this the way is should be set it would be much appreciated.

I’m just guessing based on existing perl Portfiles and don’t see the correct approach.

```
--->  Verifying Portfile for p5-net-apns-persistent
Error: Unknown dependency: p-class-accessor
Error: Unknown dependency: p-common-sense
Error: Unknown dependency: p-json-xs
Error: Unknown dependency: p-types-serialiser
--->  4 errors and 0 warnings found.
--->  Verifying Portfile for p5-privileges-drop
Error: Unknown dependency: p-http-date
Error: Unknown dependency: p-module-build
Error: Unknown dependency: p-yaml
--->  3 errors and 0 warnings found.
```

Reply:
It's only set in the versioned subports, p5.28-privileges-drop and so forth, not in p5-privileges-drop which is just a stub.

Reply:
The required change is literally just to wrap the setting of the dependencies in `if {${perl5.major} ne ""} { ... }`. One of the many examples is: https://github.com/macports/macports-ports/blob/6ff11ae4d4f9c64da34f49ac97441e51317ab66e/perl/p5-algorithm-annotate/Portfile#L21

Reply:
Great thank you! 

Okay everything is working now. I've incorporated all your comments and it builds.

Side question: I still don't understand what's going on to make this `if` block necessary, as I've already specifically specified a dependency on the `p5.28-*` ports. Why shouldn't `${perl5.major}` be set? What's a stub port?

Reply:
The portfile for p5-foo defines multiple ports. If perl5.branches is `5.28 5.30` then the portgroup code does the equivalent of this:
```
subport 5.28-foo {...}
subport 5.30-foo {...}
```
The stub port is p5-foo itself, which installs nothing but a readme file. All the code in the portfile is common to all subports, so sometimes you need to check whether it's being parsed as one of the "real" subports or the stub. Checking the value of perl5.major is a way of doing that. You could just as well check the value of $subport, but perl5.major is the convention.